shared/fs/filepreview/view-container.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+import {
+  branch,
+  compose,
+  connect,
+  mapProps,
+  renderComponent,
+  type Dispatch,
+  type TypedState,
+} from '../../util/container'
+import * as Constants from '../../constants/fs'
+import * as FsGen from '../../actions/fs-gen'
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import DefaultView from './default-view-container'
+import ImageView from './image-view'
+import TextView from './text-view'
+import {Text} from '../../common-adapters'
+
+type Props = {
+  path: Types.Path,
+  fileViewType?: Types.FileViewType, // can be set by default-view-container.js for type override
+}
+
+const mapStateToProps = (state: TypedState) => ({
+  _serverInfo: state.fs.localHTTPServerInfo,
+})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  onInvalidToken: () => dispatch(FsGen.createRefreshLocalHTTPServerInfo()),
+})
+
+const mergeProps = ({_serverInfo}, {onInvalidToken}, {path}) => ({
+  url: Constants.generateURL(path, _serverInfo.address, _serverInfo.token),
+  onInvalidToken,
+})
+
+const httpConnect = connect(mapStateToProps, mapDispatchToProps, mergeProps)
+
+export default compose(
+  mapProps(({path, fileViewType}: Props) => ({path, ft: fileViewType || Constants.viewTypeFromPath(path)})),
+  branch(({ft}) => ft === 'default', renderComponent(DefaultView)),
```

Comment:
we are really moving away from using branch for this kind of thing

Reply:
Ah OK! didn't know that. I'll replace it.